### PR TITLE
Rebrand and cleanup

### PR DESCRIPTION
**Pull request by cvium**
General cleanup. Rename API and plugin

## Dependencies before merging
- [x] #1 
- [x] https://github.com/jellyfin/jellyfin-kodi/pull/3

### CODE REVIEW

File: Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            _logger.LogDebug("Starting PopulateLibraryInfo...");
+            var userDataChangedJson = new List<string>();
+
+            var info = new SyncUpdateInfo();
+
+            var userDT = DateTime.Parse(lastDT, CultureInfo.CurrentCulture, DateTimeStyles.AssumeUniversal);
+
+            var dtl = (long)userDT.ToUniversalTime().Subtract(new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds;
+
+            _logger.LogDebug("PopulateLibraryInfo:  Getting Items Added Info...");
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs]: this logging should probably happen inside the task

File: Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                    _logger.LogInformation("Added Items Found: {AddedItems}", string.Join(",", result.ToArray()));
+                }
+                else
+                {
+                    _logger.LogInformation("No Added Items Found!");
+                }
+                return result;
+            });
+
+            _logger.LogDebug("PopulateLibraryInfo:  Getting Items Removed Info...");
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs]: same, log inside

File: Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+                    _logger.LogInformation("Removed Items Found: {RemovedItems}", string.Join(",", result.ToArray()));
+                }
+                else
+                {
+                    _logger.LogInformation("No Removed Items Found!");
+                }
+                return result;
+            });
+
+            _logger.LogDebug("PopulateLibraryInfo:  Getting Items Updated Info...");
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs]: same, log inside

File: Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                }
+                else
+                {
+                    _logger.LogInformation("No Updated Items Found!");
+                }
+                return result;
+            });
+
+            _logger.LogDebug("PopulateLibraryInfo:  Getting Folders Added To Info...");
+            info.FoldersAddedTo.Clear();
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs]: where are those used? this and `info.FoldersRemovedFrom`

File: Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            _logger.LogDebug("PopulateLibraryInfo:  Getting Folders Removed From Info...");
+            info.FoldersRemovedFrom.Clear();
+            _logger.LogDebug("PopulateLibraryInfo:  Getting User Data Changed Info...");
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs]: log inside

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs [hunk 6]
```diff
@@ -55,36 +55,31 @@ public static DbRepo Instance
...
         }
 
         public List<Guid> GetItems(long dtl, int status, bool movies, bool tvshows, bool music, bool musicvideos, bool boxsets)
         {
             var result = new List<Guid>();
             List<ItemRec> final = new List<ItemRec>();
 
-            logger.LogDebug(String.Format("Emby.Kodi.SyncQueue:  Using dtl {0:yyyy-MM-dd HH:mm:ss} for time {1}", new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc).AddSeconds(dtl), dtl));
-            logger.LogDebug(String.Format("Emby.Kodi.SyncQueue:  IntStatus: {0}", status));
+            logger.LogDebug(String.Format("Using dtl {0:yyyy-MM-dd HH:mm:ss} for time {1}", new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc).AddSeconds(dtl), dtl));
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: remove string formatting in logging (here and in other places)?

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs [hunk 7]
```diff
@@ -213,23 +208,23 @@ public void DeleteOldData(long dtl)
         {
             lock (_folderLock)
             {
-                logger.LogInformation("Emby.Kodi.SyncQueue.Task: Starting Folder Retention Deletion...");
+                logger.LogInformation("Jellyfin.Plugin.KodiSyncQueue.Task: Starting Folder Retention Deletion...");
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: why didn't you drop prefix here and in a few other places?

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 8]
```diff
@@ -237,21 +187,21 @@ private void LibraryUpdateTimerCallback(object state)
...
-                                        .Where(i => itemsAdded.Where(a => a.Id == i.Id).FirstOrDefault() == null)
+                                        .Where(i => itemsAdded.FirstOrDefault(a => a.Id == i.Id) == null)
                                         .GroupBy(g => g.Id)
                                         .Select(grp => grp.First())
                                         .ToList();
 
-                    Task x = PushChangesToDB(itemsAdded, itemsUpdated, itemsRemoved, cTokenSource.Token);
-                    Task.WaitAll(x);                    
+                    Task pushToDbTask = PushChangesToDB(itemsAdded, itemsUpdated, itemsRemoved, cTokenSource.Token);
+                    Task.WaitAll(pushToDbTask);
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs]: I don't know C# well enough, but I would expect to see `await PushChangesToDB(...)` here, why it's not that way?

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs [hunk 9]
```diff
@@ -208,20 +181,20 @@ private void UpdateTimerCallback(object state)
                 _changedItems.Clear();
                 _itemRef.Clear();
 
-                Task x = SendNotifications(changes, itemRef, cTokenSource.Token);
-                Task.WaitAll(x);
+                Task sendNotificationsTask = SendNotifications(changes, itemRef, cTokenSource.Token);
+                Task.WaitAll(sendNotificationsTask);
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs]: again, why not `await`?

File: Jellyfin.Plugin.KodiSyncQueue/Plugin.cs [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+            Instance = this;
+
+            Logger = logger;
+            Logger.LogInformation("Jellyfin.Plugin.KodiSyncQueue IS NOW STARTING!!!");
+            
+            // TODO this is just sad
+            DbRepo.dbPath = applicationPaths.DataPath;
+            DbRepo.json = json;
+            DbRepo.logger = logger;
+            DbRepo.fileSystem = fileSystem;
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Plugin.cs]: can we drop `fileSystem` here?

File: Jellyfin.Plugin.KodiSyncQueue/Plugin.cs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        private Guid _id = new Guid("b0daa30f-2e09-4083-a6ce-459d9fecdd80");
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Plugin.cs]: did you change this guid when rebranding? I think we should...

File: Jellyfin.Plugin.KodiSyncQueue/API/ServerTimeAPI.cs [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            }
+
+            if (retDays == 0)
+            {
+                retDate = new DateTime(1900, 1, 1, 0, 0, 0);
+            }
+            else
+            {
+                retDays = retDays * -1;
+                retDate = new DateTime(dtNow.Year, dtNow.Month, dtNow.Day, 0, 0, 0);
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/API/ServerTimeAPI.cs]:
```suggestion
                retDate = new DateTime(dtNow.Year, dtNow.Month, dtNow.Day, 0, 0, 0).AddDays(-retDays);
```

File: Jellyfin.Plugin.KodiSyncQueue/ScheduledTasks/RetentionTask.cs [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+            if (!int.TryParse(Plugin.Instance.Configuration.RetDays, out var retDays) || retDays == 0) {
+                _logger.LogInformation("Retention Deletion Not Possible When Retention Days = 0!");
+                return;
+            }
+
+            //Check Database
+            await Task.Run(() =>
+            {
+                retDays = retDays * -1;
+                var dt = DateTime.UtcNow.AddDays(retDays);
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/ScheduledTasks/RetentionTask.cs]:
```suggestion
                var dt = DateTime.UtcNow.AddDays(-retDays);
```
no need to multiply by `-1`

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs (hunk 8 above)
Comment (cvium) [reply Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs]: Can't await inside a lock

File: Jellyfin.Plugin.KodiSyncQueue/Plugin.cs (hunk 10 above)
Comment (cvium) [reply Jellyfin.Plugin.KodiSyncQueue/Plugin.cs]: Filesystem is still being used

File: Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+            if (!string.IsNullOrEmpty(request.filter))
+            {
+                var filter = request.filter.ToLower().Split(',');
+                foreach (var f in filter)
+                {
+                    f.Trim();
+                    switch (f)
+                    {
+                        case "movies":
+                            movies = false;
```
[the remark this answers is not in the document]
Comment (JustAMan) [reply Jellyfin.Plugin.KodiSyncQueue/API/SyncAPI.cs]: Marking as resolved as this should probably be fixed later (if at all).

**Pull request merged**